src/emeus-constraint-layout.c

```diff
@@ -1407,7 +1407,12 @@ create_child_constraint (EmeusConstraintLayout      *layout,
   /* Alternatively, if it's not a constant value, we find the variable
    * associated with it
    */
-  if (constraint->source_object != NULL)
+  if (constraint->source_object == NULL ||
+      (EMEUS_IS_CONSTRAINT_LAYOUT (constraint->source_object) && constraint->source_object == layout))
+    {
+      attr2 = get_layout_attribute (layout, constraint->source_attribute);
+    }
+  else
```

Comment:
This should be a:

```cpp
  else if (EMEUS_IS_CONSTRAINT_LAYOUT_CHILD (constraint->source_object))
    {
      ...
    }
```

So that we can add a final:

```cpp
  else
    {
      g_critical ("Invalid source object of type %s", G_OBJECT_TYPE_NAME (constraint->source_object));
      return;
    }
```